### PR TITLE
fix: Federation V2 Scalar Support

### PR DESCRIPTION
**Pull request by arunsureshkumar**
Fix : Breaks with custom scalars #13

### CODE REVIEW

File: graphene_federation/tests/test_scalar.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        parse_value = coerce_address
+
+        @staticmethod
+        def parse_literal(ast):
+            ...
+
+    class Query(ObjectType):
+        test = String(x=AddressScalar())
+
+    build_schema(query=Query, enable_federation_2=True)
```
Comment (patrick91) [inline graphene_federation/tests/test_scalar.py:26]: should we add an assert here?

File: graphene_federation/utils.py [hunk 2]
```diff
@@ -62,14 +65,20 @@ def is_valid_compound_key(type_name: str, key: str, schema: Schema):
             if field.selection_set:
                 # If the field has sub-selections, add it to node mappings to check for valid subfields
 
-                if isinstance(field_type, GraphQLScalarType):
+                if isinstance(field_type, GraphQLScalarType) or (
+                    isinstance(field_type, GraphQLNonNull)
+                    and isinstance(field_type.of_type, GraphQLScalarType)
```
Comment (patrick91) [inline graphene_federation/utils.py:70]: what about lists?

File: graphene_federation/tests/test_scalar.py (hunk 1 above)
Comment (arunsureshkumar) [reply graphene_federation/tests/test_scalar.py:26]: @patrick91, I will around this! Issue #13 could not build schema; will add an assertion on the schema!

File: graphene_federation/utils.py (hunk 2 above)
Comment (arunsureshkumar) [reply graphene_federation/utils.py:70]: > what about lists?

List will work properly, test case added!

File: graphene_federation/tests/test_scalar.py (hunk 1 above)
Comment (arunsureshkumar) [reply graphene_federation/tests/test_scalar.py]: @patrick91 - Test case modified.

**Review by patrick91 — APPROVED** [review]
Looks good, I wonder if we should rebase since there's 102 commits in the PR. But I guess we'll just squash this when merging 😊

Also /cc @erikwrede for a final check

File: examples/entities.py [hunk 3]
```diff
@@ -25,10 +33,11 @@ class Query(graphene.ObjectType):
     file = graphene.Field(File)
 
     def resolve_file(self, **kwargs):
-        return None   # no direct access
+        return None  # no direct access
 
 
 schema = build_schema(Query)
+print(schema)
```
Comment (patrick91) [inline examples/entities.py:40]: this can be removed 😊

Comment (arunsureshkumar) [reply examples/entities.py:40]: Done.

**Comment by arunsureshkumar** [comment]
> Looks good, I wonder if we should rebase since there's 102 commits in the PR. But I guess we'll just squash this when merging 😊
> 
> Also /cc @erikwrede for a final check

Yes, It's happened somehow! Please squash this when merging!

**Review by erikwrede — APPROVED** [review]

**Comment by erikwrede** [comment]
Thanks for the quick fix! ☺️

**Pull request merged**